filaments.schema.json

```diff
@@ -29,7 +29,9 @@
                         "pattern": "{"
                     },
                     "material": {
-                        "type": "string"
+                        "$comment": "Valid materials must be uppercase letters and numbers, optionally separated by + or -. Optionally ends with -CF or -GF followed by a number. Examples: ABS, ABS-CF, PC+ABS, PC+ABS-CF10.",
+                        "type": "string",
+                        "pattern": "^[A-Z0-9]+([-+][A-Z0-9]+)*$"
```

Comment:
Could make this pattern a bit strict perhaps, what do you think?
`^[A-Z0-9]+(\+[A-Z0-9]+)*(\-(CF|GF)\d*)?$`
![image](https://github.com/Donkie/SpoolmanDB/assets/2332094/3e6d81aa-7470-4c64-b793-0873a81c2290)

Reply:
I have updated the regex to be `^[A-Z0-9]+(\\+[A-Z0-9]+)*(\\-(CF|GF)\\d*)?\\+?$` so that it matches materials like `PLA+` (it was causing validation failures)